### PR TITLE
Add multi-arch head containers

### PR DESCRIPTION
**Pull request by dragonsahead**
Objective: building pure glibc multi-arch images with the official temurin images for devs

This PR will transform our CI in the following way:
1) it will build a multi-arch (amd64 and arm64) Metabase container image (both ee and oss) on every push to the main branch with GH actions
2) it will push to Dockerhub those images
3) it will remove the hook that we did some time ago specifically for Dockerhub, as now the config is passed to GH Actions
4) Stats will start running from now on a debian-based JRE container (eclipse-temurin:11.0.12_7-jre-focal)

To make everything work we need:
1) to get a token from Dockerhub and add that token to the repository secrets as DOCKERHUB_TOKEN, also add the username as DOCKERHUB_USERNAME
2) disconnect Dockerhub metabase-head and metabase-enterprise-head from Github so they don't build images anymore

We can switch to Alpine on the runner as soon as the temurin project releases multi-arch Alpine runners (pure musl buids, no glibc/musl mixtures) and also switch the runner in the release container with a simple change, as the scaffolding is ready

### CODE REVIEW

**Comment by camsaul** [comment]
Do you have a sense for how much this slows down CI runs? Or does it not make a difference?

**Review by camsaul — APPROVED** [review]
This works for me.

I'm assuming this is so we can get the perf boost on the new Apple M1s so things don't have to go thru QEMU or Rosetta 2 or whatever, right? Or do we have anyone developing on non-Apple ARM64?

A few questions if you get the chance:

- Does it affect build time significantly?
- What's the image size difference compared to the x64 Alpine images we're currently building? 
- Would we want to consider changing the runner for the Dockerfile we use for release (https://github.com/metabase/metabase/blob/master/bin/docker/Dockerfile) as well? 

If the image size difference isn't huge then maybe we should consider doing this for releases as well. AWS does have ARM64 EC2 instances these days anyway so it's not like a dev env would be the only place MB-on-ARM would ever be useful

**Comment by paoliniluis** [comment]
> This works for me.
> 
> I'm assuming this is so we can get the perf boost on the new Apple M1s so things don't have to go thru QEMU or Rosetta 2 or whatever, right? Or do we have anyone developing on non-Apple ARM64?
> 
> A few questions if you get the chance:
> 
> * Does it affect build time significantly?
> * What's the image size difference compared to the x64 Alpine images we're currently building?
> * Would we want to consider changing the runner for the Dockerfile we use for release (https://github.com/metabase/metabase/blob/master/bin/docker/Dockerfile) as well?
> 
> If the image size difference isn't huge then maybe we should consider doing this for releases as well. AWS does have ARM64 EC2 instances these days anyway so it's not like a dev env would be the only place MB-on-ARM would ever be useful

Exactly, this is a change that will allow all ARM64 machines to get a native container image without using any mechanism like Rosetta or QEMU to run. About your questions:
1) Nothing (in fact should be faster as the base image is slimmer than the one we have now coming from CircleCI)
2) Base images differ in 100MB (ubuntu vs alpine), but compressed in the repository should be around the same size
3) Not yet, I would first dogfood ourselves by switching from x86 to ARM in our own instance, and after 1-2 major versions make the change announcing that to our users.

File: .github/workflows/build-and-push-head-images.yml [hunk 1]
```diff
@@ -39,5 +39,5 @@ jobs:
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Create container image
-        run: docker buildx build . --build-arg MB_EDITION=ee -t metabase/metabase-enterprise-head:latest --push --compress --no-cache
+      - name: Create multi-arch image
+        run: docker buildx build . --build-arg MB_EDITION=ee --platform linux/arm64/v8,linux/amd64 -t metabase/metabase-enteprise-head:latest --push --compress --no-cache
```
Comment (ismith) [inline .github/workflows/build-and-push-head-images.yml:43]:
```suggestion
        run: docker buildx build . --build-arg MB_EDITION=ee --platform linux/arm64/v8,linux/amd64 -t metabase/metabase-enterprise-head:latest --push --compress --no-cache
```

Comment (paoliniluis) [reply .github/workflows/build-and-push-head-images.yml]: thanks for finding the typo!

**Comment by macnibblet** [comment]
Is there anything we can do to help get this merged? Would love to have this directly from Metabase and not rely on 3rd party images for arm support.

**Comment by paoliniluis** [comment]
I will close this PR as https://github.com/adoptium/adoptium/issues/96 seems to be a few weeks away. Once the new Alpine JRE image is up, we will be able to provide head and release images that run natively in arm64

**Comment by paoliniluis** [comment]
Note, if anyone wants to build a container for ARM, you can do it right now with the instructions I left here: https://gist.github.com/paoliniluis/78fe800f06fc7027a20665a429239c1e

**Pull request closed without merging**